Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -114,10 +134,9 @@ void DataReaderTask::run(ProcessingContext& pc)
         mTriggers = mCompressedReader.getIR();
```

Comment:
What is the reason of assigning mTracklets, mDigits, mTriggers from mCompressedReader, if in the `sendData` you reassign them again from `mReader`? 
Also, why are you doing so many copies? The `mReader` and `mCompressedReader` fill vectors, then you assign (copy) these vectors to mTracklets, mDigits, mTriggers members of this device just to create a snapshot from them. 
Why don't you do this directly from the vectors in the readers (using references)?

Reply:
It was originally pointers something was messing up, cant remember what, gave up and made it references. Was planning to come back and make it quicker. Will do this now though.

Reply:
I would avoid having these data members at all, why not creating snapshots directly from the references provided by the mCompressedReader or mReader?

Reply:
The stuff in mCompressedReader pulling the tracklets/digits/triggerrecords to a local variable was for checking purposes, and updated compressed reader is in a seperate branch to try and keep these updates logical seperate.

Reply:
> The stuff in mCompressedReader pulling the tracklets/digits/triggerrecords to a local variable was for checking purposes, and updated compressed reader is in a seperate branch to try and keep these updates logical seperate.

Sorry, did not understand what you mean, so, can we avoid extra copy?

Reply:
yes implementing in this pr.